hive/hive-export-import-metastore.md

```diff
@@ -23,73 +23,98 @@ ms.author: dkakadia
+>
+> * The procedure assumes that after completion, the old cluster will **not** be used any longer.
+
+#### Resolution Steps:
+
+If migrating from an external metastore, follow steps in [apache-hive-migrate-workloads](https://docs.microsoft.com/en-us/azure/hdinsight/interactive-query/apache-hive-migrate-workloads). Otherwise, follow steps below.
+
+1) [Connect to the HDInsight Cluster using SSH](https://docs.microsoft.com/en-us/azure/hdinsight/hdinsight-hadoop-linux-use-ssh-unix), to the primary headnode.
+
+1) Download the export script from an ssh session to the cluster:
+
+    ```bash
+    wget "https://hdiconfigactions.blob.core.windows.net/hivemetastoreschemaupgrade/$SCRIPT"
+    chmod 755 "$SCRIPT"
+    ```
+
+    where `SCRIPT="exporthive_hdi_3_6.sh"` for HDInsight 3.6 or `SCRIPT="exporthive_hdi_4_0.sh"` for HDInsight 4.0.
+
+1) Run the script from the cluster:
+
+    * For HDInsight 4.0, follow these additional steps, first:
+
+        a. Download a helper script used by `exporthive_hdi_4_0.sh`.
+
+        ```bash
+        wget "https://hdiconfigactions.blob.core.windows.net/hivemetastoreschemaupgrade/constraints2altertable.py"
+        chmod 755 constraints2altertable.py
+        ```
+
+        b. set hive.security.authorization.sqlstd.confwhitelist.append=hive.ddl.output.format in Custom hive-site via Ambari.
+
+    * For a non-ESP cluster, simply execute the script.
+    * For an ESP cluster, kinit with user with full Hive permissions, and then execute the script with modified beeline arguments:
+
+        ```bash
+        USER="USER"  # replace USER
+        DOMAIN="DOMAIN"  # replace DOMAIN
+        DOMAIN_UPPER=$(printf "%s" "$DOMAIN" | awk '{ print toupper($0) }')
+        kinit "$USER@$DOMAIN_UPPER"
+        ```
+
+        ```bash
+        hn0=$(grep hn0- /etc/hosts | xargs | cut -d' ' -f4)
+        BEE_CMD="beeline -u 'jdbc:hive2://$hn0:10001/default;principal=hive/_HOST@$DOMAIN_UPPER;auth-kerberos;transportMode=http' -n "$USER@$DOMAIN" --showHeader=false --silent=true --outputformat=tsv2 -e"
+        ./exporthive_hdi_3_6.sh "$BEE_CMD"  # replace script with exporthive_hdi_4_0.sh for 4.0
+        ```
+
+    This will generate a file named `alltables.hql`.
+
+1) Copy the file `alltables.hql` to the new HDInsight cluster and run the following command:
+
+    * For non-ESP:
+
+        ```bash
+        beeline -u "jdbc:hive2://localhost:10001/;transportMode=http" -f alltables.hql
+        ```
+
+    * for ESP:
+
+        ```bash
+        USER="USER"  # replace USER
+        DOMAIN="DOMAIN"  # replace DOMAIN
+        DOMAIN_UPPER=$(printf "%s" "$DOMAIN" | awk '{ print toupper($0) }')
+        kinit "$USER@$DOMAIN_UPPER"
+        ```
+
+        ```bash
+        hn0=$(grep hn0- /etc/hosts | xargs | cut -d' ' -f4)
+        beeline -u "jdbc:hive2://$hn0:10001/default;principal=hive/_HOST@$DOMAIN_UPPER;auth-kerberos;transportMode=http" -n "$USER@$DOMAIN" -f alltables.hql
+        ```
```

Comment:
Most of our docs have a short conclusion with a further reading/see also section at the end.
Generally speaking these docs should be framed more like articles so they are an easier read